src/infra/HttpServiceImpl.ts

```diff
@@ -39,34 +43,46 @@ export class HttpServiceImpl implements HttpService {
     url: string,
     queryParams: QueryParams,
     body: QueryParams,
+    clientSdk: Record<string, string>,
     authToken?: string
   ): Promise<T> {
     const sanitizedBodyParams = sanitizeQueryParams(body);
-    return this.fetch(url, queryParams, {
+    return fetch(url, queryParams, {
       method: RequestMethods.POST,
       body: JSON.stringify(sanitizedBodyParams),
       mode: 'cors',
       ...(authToken && { credentials: 'include' }),
       headers: {
+        'Client-SDK': formatAsHttpHeader(clientSdk),
         'Content-Type': 'application/json',
         ...(authToken && { Authorization: `Bearer ${authToken}`}),
       }
     })
       .then(res => res.json());
   }
+}
 
-  /**
-   * Perform a fetch, using the polyfill if needed.
-   */
-  private fetch(
-    url: string,
-    queryParams: QueryParams,
-    reqInit: RequestInit
-  ): Promise<Response> {
-    const urlWithParams = addParamsToURL(url, queryParams);
-    if (typeof(window) !== 'undefined' && window.fetch) {
-      return window.fetch(urlWithParams, reqInit);
-    }
-    return crossFetch(urlWithParams, reqInit);
+/**
+ * Perform a fetch, using the polyfill if needed.
+ */
+function fetch(
+  url: string,
+  queryParams: QueryParams,
+  reqInit: RequestInit
+): Promise<Response> {
+  const urlWithParams = addParamsToURL(url, queryParams);
+  if (typeof(window) !== 'undefined' && window.fetch) {
+    return window.fetch(urlWithParams, reqInit);
   }
+  return crossFetch(urlWithParams, reqInit);
 }
+
+function formatAsHttpHeader(jsonHeader: Record<string, string>) {
```

Comment:
Can we add some docs here? Also, small nit: it feels like what this method is doing is serializing a json object into an HTTP header. I think the name `jsonHeader` is a bit misleading. We're not converting one header into another. We're formatting JSON data into a header.

Reply:
Sure, I renamed `jsonHeader` to `clientSdk` and added docs with an example that should make it more clear.